src/server.cpp

```diff
@@ -23,7 +23,7 @@ const char * inFifoSemaphoreName = "/serverInFifoSemaphore";
 GuardedQueue writeReqQueue;
 GuardedQueue readReqQueue;
 sem_t * inputFifoSemaphore;
-std::unordered_set<Tuple> tupleSpace;	// Global tuple space
+multimap<size_t, Tuple> tupleSpace; //Global tuple space
```

Comment:
As far as I read std::unordered_set complexity is better than std::multimap. std::multimap is sorted map and hence it must maintain its order and operations take O(logn) while in std::unordered_set it is mostly O(1)

Reply:
Yes, but set provides uniqueness, so it's not only checking hash values but also equality of items, in our case we can have more than 1 same tuple and changing '==' operator for Tuple is logically incorrect, but yes, ordering is not needed, i'll change to unordered_multimap

Reply:
ok, unordered_multimap is very good